### PR TITLE
GEN-1983 : Feat -   Support Tag Asset Page

### PR DESCRIPTION
**Pull request by sonika-shah**
### Describe your changes:

Feat :  Support bulk adding tag to Assets - async operation 
Fix : #15541

https://github.com/user-attachments/assets/521f9c7b-9511-4137-ad70-b7dd7bfab9de

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TagRepository.java [hunk 1]
```diff
@@ -109,6 +122,82 @@ public void setFullyQualifiedName(Tag tag) {
     }
   }
 
+  public BulkOperationResult bulkAddAndValidateTagToAssets(
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TagRepository.java:125]: is this async operation?

Comment (sonika-shah) [reply openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TagRepository.java:125]: @harshach , this is sync operation , similar to the current implementation that we have for adding assets to glossaryTerm on Assets tab

image

**Comment by harshach** [comment]
@sonika-shah we should make this async operation similar to what we have done in exportCSV. 
all of these bulk operations will run into same issue when there are lot of assets being tagged.
IMO its better to use our executorService to make these ops async and use websockets to send the notifications back

**Review by harshach — APPROVED** [review]

**Comment by harshach** [comment]
@sonika-shah did we do testing with large amounts of data?

**Comment by sonika-shah** [comment]
@harshach , ui is still making progress here, once it gets completed , will cover the testing part too

**Review by mohityadav766 — APPROVED** [review]

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**